src/test/java/service/impl/WriterServiceImplTest.java

```diff
@@ -0,0 +1,69 @@
+package service.impl;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import dao.FruitStorageDao;
+import dao.FruitStorageDaoImpl;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.List;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import service.ReaderService;
+import service.WriterService;
+
+class WriterServiceImplTest {
+    private static final String TEST_FILE = "src/main/resources/test.csv";
+    private WriterService writer;
+    private FruitStorageDao fruitStorageDao;
+    private ReaderService reader;
+    private Path testFilePath;
+
+    @BeforeEach
+    void setUp() {
```

Comment:
Don't use try catch throw in unit tests, use assertThrows if you expect exception. Unexpected exception _will_ fail the test, no need to rethrow anything, this isn't business logic (where we catch something specific and wrap it in our customary exception)